zerver/lib/bugdown/__init__.py

```diff
@@ -1589,6 +1595,14 @@ def __init__(self, realm_id, content):
 
         user_group_names = possible_user_group_mentions(content)
         self.user_group_name_info = get_user_group_name_info(realm_id, user_group_names)
+        self.user_group_members = {}  # type: Dict[Text, List[int]]
+        membership = UserGroupMembership.objects.all()
+        for group_name in self.user_group_name_info:
+            members = [
+                member.user_profile.id
+                for member in membership.filter(user_group__name=group_name)
+            ]
+            self.user_group_members[group_name.lower()] = members
```

Comment:
Two problems with this block:
(1) `UserGroupMembership.objects.all()` is wrong -- we should always filter on realm for this sort of thing, and this does database queries in a loop (if there are 100 groups: 100 queries).
(2) I think we want to do this collapsing of the two states in the `do_send_messages` layer, not in Bugdown, since eventually we may want these to do something different.   So I think it'd be better to just modify `mentioned_user_ids` in `do_send_messages` to be extended by the list of user IDs in a mentioned group (which I think we could do with `set(UserMembership.objects.filter(user_group__in=user_group_ids).values("user_profile_id"))`.

Reply:
I have used the "IN" query, but we don't have the `realm` field in this table.